### ISSUE TITLE
WinAppDriver Not Finding Certain Elements[Specifically Items under Telerik RabTabControl]

### ISSUE THREAD

**Issue by pavit84** [microsoft/WinAppDriver#1707]
Trying to find the element Connect[attached below] but WinAppDriver ended up not finding the element for some. Upon checking the pagesource found out the pagesource does not have it even though its visible after the app is launched via test. The Logon Tab Item in the screenshot attached[first screenshot] is a Telerik RadTabControl.

_These are some of the things have tried :_
1. Checked that there is only one window handle so does not seem like related to multiple handles open and not switching to the correct one is an issue here. 
2. Have also tried using different locators to find the element[Name and Accessibility ID].
3. The element is visible on the screen always when application is launched still tried waiting for an element to see if that resolves the issue.
![image](https://user-images.githubusercontent.com/25206942/163825825-cea48a0e-51eb-47f8-bd1e-069d7d9ba31a.png)

**Note:** Sometimes Inspect.exe the locator tool we are using has the element but sometimes have to refresh it multiple times so that element is visible for us to grab the locator value. 

Currently the solution that works for us is to click on the element[ i.e. btnLogon  highlighted in the second screenshot ] till we get the btnConnect in the pagesource[use while look to check the pagesource] but not satisfied with this solution.

![image](https://user-images.githubusercontent.com/25206942/163827563-dd508d4d-f3be-4494-ae96-4ef9d97bb182.png)

Same issue is seen for Telerik RadGridView specifically when trying to get grid's data. 

Any advise on what we are doing wrong.

**Comment by Urvika-gola** [comment]
For the issue have you tried using WinAppDriverUIRecorder?
_"Note: Sometimes Inspect.exe the locator tool we are using has the element but sometimes have to refresh it multiple times so that element is visible for us to grab the locator value. The custom node sometimes does not show up during first arrival to the app but then returning back make the custom node appear and so as the items."_

**Comment by anunay1** [comment]
@pavit84 I am not able to figure out the issue, if possible you can send a zoom call invite anunayathakur1@gmail.com.

**Comment by anunay1** [comment]
and I see that you are switching window handle: 
String strPageSource = Session.SwitchTo().Window(Session.WindowHandles[0]).PageSource;

what's the reason behind it?

**Comment by pavit84** [comment]
yes. With the switch was trying to refresh the page and then get the Pagesource to see if it helps me getting the full Pagesource.I am able to see the UI but the Pagesource does not contain all the elements even though visible on the screen. Will send you the invite what time works for you? The only thing or the solution that works for me is clicking the logon button multiple times till the pagesource contains the connect button(while loop).

**Comment by anunay1** [comment]
You can send the invite for tomorrow 7 PM IST?

**Comment by anunay1** [comment]
That should be fine with me
